diesel_tests/tests/types_roundtrip.rs

```diff
@@ -344,6 +348,60 @@ pub fn mk_naive_date(days: u32) -> NaiveDate {
     earliest_sqlite_date + Duration::days(days as i64 % num_days_representable)
 }
 
+#[cfg(any(feature = "postgres", feature = "mysql"))]
+#[derive(Clone, Debug)]
+struct SerdeWrapper(serde_json::Value);
+
+#[cfg(any(feature = "postgres", feature = "mysql"))]
+impl quickcheck::Arbitrary for SerdeWrapper {
+    fn arbitrary<G: quickcheck::Gen>(g: &mut G) -> Self {
+        SerdeWrapper(arbitrary_serde(g, 0))
+    }
+}
+
+#[cfg(any(feature = "postgres", feature = "mysql"))]
+fn arbitrary_serde<G: quickcheck::Gen>(g: &mut G, depth: usize) -> serde_json::Value {
+    use rand::distributions::Alphanumeric;
+    use rand::Rng;
+    match g.gen_range(0, if depth > 0 { 4 } else { 6 }) {
```

Comment:
Shouldn't the `4` and `6` be reverse? Or am I missing something?

Reply:
I think it's correct this way, because:

* It works 
* It is used to prevent endless recursion, so that we just generate objects and lists for the first level, not for anything else. That means if `depth > 0`, i.e. we've already recursed at least once we only generate variants that cannot recurse again.

(The second constrain can be lifted to allow some more nesting)